core/src/main/scala/pureconfig/ConfigConvert.scala

```diff
@@ -155,27 +157,42 @@ object ConfigConvert extends LowPriorityConfigConvertImplicits {
   case class NoValidCoproductChoiceFound(config: ConfigValue)
     extends RuntimeException(s"No valid coproduct type choice found for configuration $config.")
 
-  implicit def cNilConfigConvert: ConfigConvert[CNil] = new ConfigConvert[CNil] {
+  implicit def cNilConfigConvert[Wrapped]: WrappedConfigConvert[Wrapped, CNil] = new WrappedConfigConvert[Wrapped, CNil] {
     override def from(config: ConfigValue): Try[CNil] =
       Failure(NoValidCoproductChoiceFound(config))
 
     override def to(t: CNil): ConfigValue = ConfigFactory.parseMap(Map().asJava).root()
   }
 
-  implicit def coproductConfigConvert[V, T <: Coproduct](
-    implicit vFieldConvert: Lazy[ConfigConvert[V]],
-    tConfigConvert: Lazy[ConfigConvert[T]]): ConfigConvert[V :+: T] =
-    new ConfigConvert[V :+: T] {
+  implicit def coproductConfigConvert[Wrapped, Name <: Symbol, V, T <: Coproduct](
+    implicit coproductHint: CoproductHint[Wrapped],
+    vName: Witness.Aux[Name],
+    vFieldConvert: Lazy[ConfigConvert[V]],
+    tConfigConvert: Lazy[WrappedConfigConvert[Wrapped, T]]): WrappedConfigConvert[Wrapped, FieldType[Name, V] :+: T] =
+    new WrappedConfigConvert[Wrapped, FieldType[Name, V]:+: T] {
+
+      override def from(config: ConfigValue): Try[FieldType[Name, V] :+: T] = {
```

Comment:
I think a missing key should be handled consistently across the implicit derivations PureConfig provides. If we allow `CoproductHint` instances to read nulls, a missing key that points to a coproduct type can be handled differently by the coproduct `ConfigConvert`, and I think that decision should be left to the enclosing object's `ConfigConvert`. For instance, if I have:

```scala
case class Conf(animal: AnimalConfig)
pureconfig.loadConfig[Conf](ConfigFactory.empty)
```

I'm expecting to see a `KeyNotFoundException("animal")`, regardless of the `ConfigConvert` that is being implicitly provided for `AnimalConfig`. If the key is explicitly set to `null` in the configuration file, then we're operating on a `ConfigNull` and we can have a special treatment for it in the `CoproductHint`.